connectors/aws/aws-sagemaker/src/main/java/io/camunda/connector/sagemaker/model/SageMakerSyncResponse.java

```diff
@@ -7,16 +7,37 @@
 package io.camunda.connector.sagemaker.model;
 
 import com.amazonaws.services.sagemakerruntime.model.InvokeEndpointResult;
+import io.camunda.connector.aws.ObjectMapperSupplier;
+import java.io.IOException;
+import java.nio.ByteBuffer;
 import java.nio.charset.StandardCharsets;
 
 public record SageMakerSyncResponse(
-    String body, String contentType, String customAttributes, String invokedProductionVariant) {
+    Object body, String contentType, String customAttributes, String invokedProductionVariant) {
 
   public SageMakerSyncResponse(InvokeEndpointResult result) {
     this(
-        StandardCharsets.UTF_8.decode(result.getBody()).toString(),
+        mapResponseBody(result),
         result.getContentType(),
         result.getCustomAttributes(),
         result.getInvokedProductionVariant());
   }
+
+  /* Maps the response body to the appropriate object type
+  / https://docs.aws.amazon.com/sagemaker/latest/dg/clarify-online-explainability-invoke-endpoint.html#clarify-online-explainability-response
+  */
+  private static Object mapResponseBody(InvokeEndpointResult result) {
+    if (result.getContentType().equals("application/json")) {
```

Comment:
I don't know the `result` var, but if it can be something like `application/json; charset=utf-8` we might want to : 
1. lower case 
2. startsWith instead of equals
wdyt?

Reply:
also wondering if we should add that in the SDK, it looks like we're doing this comparison in a lot of different places

Reply:
@johnBgood I based it on this documentation: https://docs.aws.amazon.com/sagemaker/latest/dg/clarify-online-explainability-invoke-endpoint.html#clarify-online-explainability-response
  
> also wondering if we should add that in the SDK, it looks like we're doing this comparison in a lot of different places

Potentially yes but the implementation details might still be different based on the API.